client/faucet/src/mint.rs

```diff
@@ -109,83 +64,114 @@ impl MintParams {
             client,
         }
     }
 
-    pub async fn process(&self, params: &MintParams) -> Result<Response> {
+    pub async fn process(&self, mut params: MintParams) -> Result<Response> {
         let (tc_seq, dd_seq, receiver_seq) = self.sequences(params.receiver()).await?;
 
-        let mut txns = vec![];
-        if receiver_seq.is_none() {
-            let script = if params.is_designated_dealer.unwrap_or(false) {
-                params.create_designated_dealer_account_script(tc_seq)
-            } else {
-                params.create_parent_vasp_account_script(tc_seq)
-            };
-            txns.push(self.create_txn(script, treasury_compliance_account_address(), tc_seq)?);
-        }
-        txns.push(self.create_txn(params.p2p_script(), testnet_dd_account_address(), dd_seq)?);
+        let txns = {
+            let mut treasury_account = self.treasury_account.lock().unwrap();
+            let mut dd_account = self.dd_account.lock().unwrap();
+
+            // If the onchain sequence_number is greater than what we have, update our
+            // sequence_numbers
+            if tc_seq > treasury_account.sequence_number() {
+                *treasury_account.sequence_number_mut() = tc_seq;
+            }
+            if dd_seq > dd_account.sequence_number() {
+                *dd_account.sequence_number_mut() = tc_seq;
+            }
+
+            let mut txns = vec![];
+            if receiver_seq.is_none() {
+                let builder = if params.is_designated_dealer.unwrap_or(false) {
+                    self.transaction_factory.create_designated_dealer(
+                        params.currency_code,
+                        0, // sliding_nonce
+                        params.auth_key,
+                        &format!("No. {} DD", treasury_account.sequence_number()),
+                        false, // add all currencies
+                    )
+                } else {
+                    self.transaction_factory.create_parent_vasp_account(
+                        params.currency_code,
+                        0, // sliding_nonce
+                        params.auth_key,
+                        &format!("No. {} VASP", treasury_account.sequence_number()),
+                        false, // add all currencies
+                    )
+                };
+
+                txns.push(treasury_account.sign_with_transaction_builder(builder));
+            }
+
+            txns.push(dd_account.sign_with_transaction_builder(
+                self.transaction_factory.peer_to_peer_with_metadata(
+                    params.currency_code,
+                    params.receiver(),
+                    params.amount,
+                    params.bcs_metadata(),
+                    vec![],
+                ),
+            ));
+            txns
+        };
 
         let batch = txns
             .iter()
             .map(|txn| MethodRequest::submit(txn))
             .collect::<Result<_, _>>()?;
```

Comment:
Nit, maybe unwrap is better as we don't expect an error here and panic will be better option if it happened